src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,33 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || this.getClass() != obj.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) obj;
+        return (this.width == null && rectangle.width == null ? true :
+                this.width != null ? this.width.equals(rectangle.width) : false)
+                &&
+                (this.length == null && rectangle.length == null ? true :
+                        this.length != null ? this.length.equals(rectangle.length) : false)
+                &&
+                (this.color == null && rectangle.color == null ? true :
+                        this.color != null ? this.color.equals(rectangle.color) : false);
+    }
```

Comment:
remove this keyword

Reply:
https://mateacademy.slack.com/archives/G01EHB8CWAE/p1607422728299900

Reply:
I see the 3 rules: 
- same name in method parms and class fiels
- if we pass to over object 
- to call over constructor in constructor

Reply:
done